### ISSUE TITLE
Slightly more refined option type equality

### ISSUE THREAD

**Issue by Zdancewic** [upenn-cis1xx/camelot#54]
The current default linter settings give warnings for test cases that involve options.  For example the following code generates a warning from the comparison.
```
(* access value for key in the map *)
let test () : bool =
   (get 1 m1) = Some "uno"
;; run_test "find 1 m1" test
```
It would be nice to issue these warnings only when the `e = Some _` expression appears in an `if` guard.

**Comment by Vighnesh-V** [comment]
Over the summer we ran into an issue like this in HW2, so we wrote a small config language for the linter which allows us to disable rules at a global level or at a specific named top-level.

So for an example like this, we probably wouldn't want them checking option type equality in the stub code they have to fill out for the homework, but we would want to disable the rule for the testing code they write.

We can write a linter config file:
```
{
    "toLint": ["./file.ml"],
    "global": {
         "disable": []
    }, 
    "locals": [ 
      { "test": { "disable": ["EqOption"] } }
    ]
}
```
which will disable the option equality rule for top-level functions called test.

This should still flag when students write things like `if e = Some x ...` in top-level functions not called `test`, since we apply the rules recursively and the EqOption rule tries to find instances of just `e = Some x` .

I think by default the lint config file is called `arthur.json` and we look for it in the current directory where `camelot` is invoked from. Otherwise I think the `-c` argument should let you specify a path to lint config file you want :) .

During the summer we just added the lint config file to the codio box. For example, in HW2, the `arthur.json` looks a like:
```
{
    "toLint": ["./dna.ml", "./dna.mli", "./dnaTest.ml"],
    "global": {
         "disable": ["UseMap", "UseIter", "UseFold"]
    }, 
    "locals": [ 
      { "string_of_ape": { "disable": ["NestedIf"] } }
    ]
}
```
